docs/specifications/tests/DNSSEC-TP/dnssec11.md

```diff
@@ -1,50 +1,244 @@
+
+       1. Send *DS Query* to the name server IP.
+       1. If the response has the TC flag set, re-query over TCP and use that
+          response instead.
+       2. If there is no DNS response, then add the name server (IP) to the
+          *Indetermined DS* set.
+       3. Else, if the RCODE of response is not "NoError" ([IANA RCODE List]),
+          then add the name server (IP) to the *Indetermined DS* set.
+       4. Else, if the AA flag is not set in the response, then add the name
+          server (IP) to the *Indetermined DS* set.
+       5. Else, if there is no DS record with matching owner name in the
+          answer section, then add the name server (IP) to the
+          *No DS Record* set.
+       7. Else add the name server (IP) to the *Has DS Record* set.
+
+    4. If the *Indetermined DS* set is non-empty and both the
+       *No DS Record* and *Has DS Record* sets are empty then do:
+       1. Output *[DS11_INDETERMINED_DS]*.
+       2. Exit this test case.
+
+    5. If the *No DS Record* set is non-empty and the *Has DS Record* set is
+       empty then exit this test case.
+
+    6. If both the *No DS Record* and *Has DS Record* sets are non-empty,
+       then do:
+       1. Output *[DS11_INCONSISTENT_DS]*.
+       2. Output *[DS11_PARENT_WITHOUT_DS]* and list parent name servers in
+          *No DS Record*.
+       3. Output *[DS11_PARENT_WITH_DS]* and list parent name servers in
+          *Has DS Record*.
+
+3.  If the *Test Type* is "[undelegated]" and if *Undelegated DS* is empty,
+    then do exit this test case.
+
+4.  Create DNS queries for the child zone:
+
+    1.  SOA query for the *Child Zone* without any OPT record (no EDNS)
+        ("SOA Query").
+    2.  DNSKEY query for the *Child Zone* with the DO flag set ("DNSKEY Query").
+
+5.  Obtain the set of child name server IP addresses using [Method4] and
+    [Method5] ("Child NS IP").
 
-### Intercase dependencies
+7.  For each name server IP in *Child NS IP* do:
+
+    1. Send *SOA Query* over UDP to the name server IP and collect the response.
+    2. Go to next name server if
+       1. there is no DNS response on the SOA query, or
+       2. the RCODE of the response is not "NoError" ([IANA RCODE List]), or
+       3. the AA flag is not set in the response, or
+       4. there is no SOA record with owner name matching the query in the 
+          answer section.
+
+    3. Send *DNSKEY Query* over UDP to the name server IP and collect the
+       response.
+    4. If the response has the TC flag set, re-query over TCP and use that
+       response instead.
+    5. If there is no DNS response, then add the name server (IP) to the
+       *Indeterminded DNSKEY* set.
+    6. Else, if the RCODE of response is not "NoError" ([IANA RCODE List]),
+       then add the name server (IP) to the *Indeterminded DNSKEY* set.
+    7. Else, if the AA flag is not set in the response, then add the name server
+       IP to the *Indeterminded DNSKEY* set.
```

Comment:
these can be grouped in a sub listing.

Reply:
It does not work well. See above.

Reply:
Again, I would argue that the same action is performed in either cases.

Reply:
I will group.